src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1510,3 +1730,163 @@ void java_bytecode_parsert::parse_local_variable_type_table(methodt &method)
       "Entry in LocalVariableTypeTable must be present in LVT");
   }
 }
+
+
+
+/// Read method handle pointed to from constant pool entry at index, return type
+/// of method handle and name if lambda function is found.
+/// \param entry: the constant pool entry of the methodhandle_info structure
+/// \returns: the method_handle type of the methodhandle_structure,
+/// either for a recognized bootstrap method or for a lambda function
+optionalt<java_bytecode_parsert::lambda_method_handlet>
+java_bytecode_parsert::parse_method_handle(const method_handle_infot &entry)
+{
+  const std::function<pool_entryt &(u2)> pool_entry_lambda =
+    [this](u2 index) -> pool_entryt & { return pool_entry(index); };
+
+  const base_ref_infot &ref_entry = entry.get_reference(pool_entry_lambda);
+
+  const class_infot &class_entry=ref_entry.get_class(pool_entry_lambda);
+  const name_and_type_infot &name_and_type =
+    ref_entry.get_name_and_type(pool_entry_lambda);
+
+  const std::string method_name =
+    class_entry.get_name(pool_entry_lambda) + "." +
+    name_and_type.get_name(pool_entry_lambda) +
+    name_and_type.get_descriptor(pool_entry_lambda);
+
+  lambda_method_handlet lambda_method_handle;
+
+  if(
+    has_prefix(name_and_type.get_name(pool_entry_lambda), "lambda$"))
+  {
+    // names seem to be lambda$POSTFIX$NUM
+    // where POSTFIX is FUN for a function name in which the lambda is define
+    //                "static" when it is a static member of the class
+    //                "new" when it is a class variable, instantiated in <init>
+    lambda_method_handle.lambda_method_name =
+      name_and_type.get_name(pool_entry_lambda);
+    lambda_method_handle.handle_type =
+      method_handle_typet::LAMBDA_METHOD_HANDLE;
+
+    return lambda_method_handle;
+  }
+
+  return {};
+}
+
+/// Read all entries of the `BootstrapMethods` attribute of a class file.
+/// \param parsed_class: the class representation of the class file that is
+/// currently parsed
+void java_bytecode_parsert::read_bootstrapmethods_entry(
+  classt &parsed_class)
+{
+  u2 num_bootstrap_methods = read_u2();
+  for(size_t i = 0; i < num_bootstrap_methods; i++)
+  {
+    u2 bootstrap_methodhandle_ref = read_u2();
+    const pool_entryt &entry = pool_entry(bootstrap_methodhandle_ref);
+
+    method_handle_infot method_handle{entry};
+
+    u2 num_bootstrap_arguments = read_u2();
```

Comment:
I still think it would be better to read all the args into a `std::vector<u2>` to avoid risk missing an arg if we add more special cases to the different bootstrap methods we understand